src/fr/poseidonscans/src/eu/kanade/tachiyomi/extension/fr/poseidonscans/PoseidonScans.kt

```diff
@@ -1,14 +1,569 @@
+            "$baseUrl/series"
+        }
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val requestUrl = response.request.url
+        val searchQuery = requestUrl.queryParameter("app_query")?.takeIf { it.isNotBlank() } ?: ""
+
+        val pageDataJson = extractNextJsPageData(document)
+            ?: return MangasPage(emptyList(), false)
+
+        val mangaListJsonArray = pageDataJson["mangas"]?.jsonArray
+            ?: pageDataJson["series"]?.jsonArray
+            ?: pageDataJson["initialData"]?.jsonObject?.get("mangas")?.jsonArray
+            ?: pageDataJson["initialData"]?.jsonObject?.get("series")?.jsonArray
+            ?: return MangasPage(emptyList(), false)
+
+        val allMangas = mangaListJsonArray.mapNotNull { mangaElement ->
+            try {
+                val mangaObject = mangaElement.jsonObject
+                val title = mangaObject["title"]?.jsonPrimitive?.content ?: return@mapNotNull null
+                val slug = mangaObject["slug"]?.jsonPrimitive?.content ?: return@mapNotNull null
+                val cover = mangaObject["coverImage"]?.jsonPrimitive?.content
+
+                SManga.create().apply {
+                    this.title = title
+                    this.url = "/serie/$slug"
+                    this.thumbnail_url = cover?.takeIf { it.isNotBlank() }?.toApiCoverUrl()
+                }
+            } catch (e: Exception) {
+                null
+            }
+        }
+
+        val filteredMangas = if (searchQuery.isNotBlank()) {
+            allMangas.filter { manga ->
+                manga.title.contains(searchQuery, ignoreCase = true)
+            }
+        } else {
+            allMangas
+        }
+
+        // /series loads all items at once (client-side 'load more'), so no next page from this specific request.
+        val hasNextPage = false
+        return MangasPage(filteredMangas, hasNextPage)
+    }
+
+    override fun latestUpdatesSelector() = throw UnsupportedOperationException("Uses JSON API")
+    override fun latestUpdatesFromElement(element: Element) = throw UnsupportedOperationException("Uses JSON API")
+    override fun latestUpdatesNextPageSelector() = throw UnsupportedOperationException("Uses JSON API")
+
+    override fun chapterListSelector() = throw UnsupportedOperationException("Uses JSON")
+    override fun chapterFromElement(element: Element) = throw UnsupportedOperationException("Uses JSON")
+
+    override fun imageUrlParse(document: Document): String = throw UnsupportedOperationException("Uses pageListParse")
+
+    override fun searchMangaSelector(): String = throw UnsupportedOperationException("Uses JSON")
+    override fun searchMangaFromElement(element: Element): SManga = throw UnsupportedOperationException("Uses JSON")
+    override fun searchMangaNextPageSelector(): String? = throw UnsupportedOperationException("No URL pagination for search")
+
```

Comment:
consider inheriting from `HttpSource` as you are not using most of the `ParsedHttpSource` methods